### PR TITLE
[shelly] Shelly Motion, minor improvements

### PR DESCRIPTION
**Pull request by markus7017**
This PR adds support for Shelly Motion and includes minor improvements

### CODE REVIEW

**Comment by markus7017** [comment]
@fwolter reported issues from hardening have been fixed, we are ready for review

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java [hunk 1]
```diff
@@ -145,6 +145,7 @@ public boolean handleDeviceCommand(ChannelUID channelUID, Command command) throw
     }
 
     /**
+     * PaperUI Control has a combined Slider for Brightness combined with On/Off
```
Comment (cpmeister) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java:148]: I don't think we should mention PaperUI anymore.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyDeviceStats.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.Map;
+
+import org.openhab.binding.shelly.internal.util.ShellyUtils;
+
+/***
+ * {@link ShellyDeviceStats} some statistical values for the thing
+ *
+ * @author Markus Michels - Initial contribution
+ */
+public class ShellyDeviceStats {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyDeviceStats.java:25]: Please add `@NonNullByDefault`

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java [hunk 3]
```diff
@@ -154,8 +159,38 @@ public void processResponse(@Nullable Response response) {
...
+                        // Format: <device type>#<mac address>#<coap version>
+                        String macid = substringBetween(devid, "#", "#");
+                        if (profile.mac.toUpperCase().contains(macid.toUpperCase())) {
+                            match = true;
+                            break;
+                        }
+                    }
+                }
+                i++;
+            }
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java:190]: changed

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java [hunk 4]
```diff
@@ -173,84 +208,79 @@ public void processResponse(@Nullable Response response) {
...
-                                } else {
-                                    logger.warn("{}: Unsupported CoAP version detected: {}", thingName, sVersion);
-                                    return;
-                                }
-                                coiotVers = iVersion;
-                                coiotBound = true;
+            payload = response.getPayloadString();
+            int i = 0;
+            while (i < options.size()) {
+                Option opt = options.get(i);
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java:214]: changed

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java:148]: changed

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyDeviceStats.java (hunk 2 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyDeviceStats.java:25]: changed

**Comment by markus7017** [comment]
@cpmeister changes applied, ready to merge

**Comment by markus7017** [comment]
@cpmeister is there anything open?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java [hunk 5]
```diff
@@ -164,93 +200,88 @@ public void processResponse(@Nullable Response response) {
...
-                            logger.debug("{} ({}): COAP option {} with value {} skipped", thingName, devId,
-                                    opt.getNumber(), opt.getValue());
-                    }
-                    i++;
+                            coiotVers = iVersion;
+                            coiotBound = true;
+                        }
+                        break;
+                    case COIOT_OPTION_STATUS_VALIDITY:
+                        // validity = o.getIntegerValue();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java:250]: Is this no longer needed? If so please remove it.

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java (hunk 5 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java:250]: removed

**Comment by markus7017** [comment]
@cpmeister change is committed

**Comment by markus7017** [comment]
@cpmeister could we please complete the merge, I already have the next PR ready to submit

**Comment by markus7017** [comment]
@cpmeister anything missing?

**Comment by markus7017** [comment]
@fwolter could you help here? It seems that @cpmeister is pretty busy

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapServer.java [hunk 6]
```diff
@@ -49,7 +49,7 @@
     boolean started = false;
     private CoapEndpoint statusEndpoint = new CoapEndpoint.Builder().build();
     private @Nullable UdpMulticastConnector statusConnector;
-    private final CoapServer server = new CoapServer(NetworkConfig.getStandard(), COIOT_PORT);;
+    private CoapServer server = new CoapServer(NetworkConfig.getStandard(), COIOT_PORT);;
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapServer.java:52]: changed

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java [hunk 7]
```diff
@@ -244,7 +244,8 @@ private void handleRoller(Command command, String groupName, Integer index, bool
                     logger.debug("{}: Use favoriteUP id {} for positioning roller({}%)", thingName, config.favoriteUP,
                             pos);
                 }
-            } else if ((command == UpDownType.DOWN) || (command == OnOffType.OFF)) {
+            }
+            if ((command == UpDownType.DOWN) || ((command == OnOffType.OFF))) {
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyRelayHandler.java:248]: changed

**Comment by markus7017** [comment]
@fwolter changes committed. I also included a fix for H&T and Motion channel creation

**Review by fwolter — APPROVED** [review]
LGTM

There are some checkstyle warnings, which can be fixed easily. Can you do this in some follow-up PR?

**Comment by markus7017** [comment]
@fwolter thanks a lot

**Comment by markus7017** [comment]
@fwolter 

fyi: 
```
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerPage.java:[32]
The package org.apache.commons.io.IOUtils should not be used.
```
has already be solved

how to fix those warnings?
```
[WARNING] org.openhab.binding.shelly.internal.provider.ShellyChannelDefinitions.java:[480]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.provider.ShellyChannelDefinitions.java:[581]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerPage.java:[84]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerPage.java:[119]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerPage.java:[125]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerPage.java:[129]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.manager.ShellyManagerCache.java:[78]
Classes/Interfaces should be annotated with @NonNullByDefault
[WARNING] org.openhab.binding.shelly.internal.coap.ShellyCoapServer.java:[55]
Classes/Interfaces should be annotated with @NonNullByDefault
```

That refers to inner classes. I know that this check is disabled for *DTO.java, but how to convince the style checker on inner classes in regular class files?

**Comment by fwolter** [comment]
I don't get any of these warnings. Did you run `mvn clean install`?

**Pull request merged**